src/core_modules/capture-core/components/WidgetsChangelog/common/hooks/useClientDataValues.js

```diff
@@ -0,0 +1,188 @@
+import { errorCreator, buildUrl, pipe } from 'capture-core-utils';
+import { ReactQueryAppNamespace } from 'capture-core/utils/reactQueryHelpers';
+import { dataElementTypes } from '../../../../metaData';
+import { CHANGELOG_ENTITY_TYPES } from '../Changelog/Changelog.constants';
+import type { Change, ItemDefinitions, SortDirection } from '../Changelog/Changelog.types';
+import { convertServerToClient, convertClientToView } from '../../../../converters';
+import { convert } from '../../../../converters/clientToList';
+import { RECORD_TYPE, subValueGetterByElementType } from '../utils/getSubValueForChangelogData';
+import { makeQuerySingleResource } from '../../../../utils/api';
+
+const convertFn = pipe(convertServerToClient, convertClientToView);
+
+type Props = {
+    rawRecords: Object,
+    dataItemDefinitions: ItemDefinitions,
+    entityId: string,
+    entityType: $Values<typeof CHANGELOG_ENTITY_TYPES>,
+    programId?: string,
+    sortDirection: SortDirection,
+    page: number,
+    pageSize: number,
+};
+
+const fetchFormattedValues = async ({
+    rawRecords,
+    dataItemDefinitions,
+    entityId,
+    entityType,
+    programId,
+    absoluteApiPath,
+    querySingleResource,
+    fromServerDate,
+}) => {
+    if (!rawRecords) return [];
+
+    const mostRecentCreatedAtByFieldId = rawRecords.changeLogs.reduce((acc, record) => {
+        const elementKey = Object.keys(record.change)[0];
+        const fieldId = record.change[elementKey]?.dataElement ?? record.change[elementKey]?.attribute;
+
+        if (!acc[fieldId] || moment(record.createdAt).isAfter(acc[fieldId])) {
+            acc[fieldId] = record.createdAt;
+        }
+        return acc;
+    }, {});
+
+    const getMetadataItemDefinition = (
+        elementKey: string,
+        change: Change,
+        dataItem: ItemDefinitions,
+    ) => {
+        const { dataElement, attribute } = change;
+        const fieldId = dataElement ?? attribute;
+        const metadataElement = fieldId ? dataItem[fieldId] : dataItem[elementKey];
+        return { metadataElement, fieldId };
+    };
+
+    const fetchedRecords = await Promise.all(
+        rawRecords.changeLogs.map(async (changelog) => {
+            const { change: apiChange, createdAt, createdBy, type } = changelog;
+            const elementKey = Object.keys(apiChange)[0];
+            const change = apiChange[elementKey];
+
+            const { metadataElement, fieldId } = getMetadataItemDefinition(
+                elementKey,
+                change,
+                dataItemDefinitions,
+            );
+
+            if (!metadataElement) {
+                log.error(
+                    errorCreator('Could not find metadata for element')({ ...changelog }),
+                );
+                return null;
+            }
+
+            const getSubValue =
+                subValueGetterByElementType[RECORD_TYPE[entityType]]?.[metadataElement.type];
+
+            const isLatestValue =
```

Comment:
The isLatestValue logic works fine for the first page, but behaves strangely on the other pages. For example: 
- on page 1 the IMAGE is deleted
- on page 2 the IMAGE is created

When navigating to page 2, the isLatestValue of the now-deleted image is calculated to be true. Because the image has already been deleted, there is nothing to display and a broken image preview is shown. Clicking on it takes the user to a 404 error page.

https://github.com/user-attachments/assets/18a7a25a-a30f-4bd3-a578-499dab99b285




Can you have a look @henrikmv? Let me know if you need any help! Thank you!

Reply:
@simonadomnisoru After discussions with @JoakimSM, I have now compared the changelog data to the event/tracked entity data to find out which image or file is the latest. Do you agree?